service/javadsl/api/src/main/java/com/lightbend/lagom/javadsl/api/ServiceInfo.java

```diff
@@ -3,18 +3,79 @@
  */
 package com.lightbend.lagom.javadsl.api;
 
+import java.util.*;
+
 /**
- * Information for this service.
+ * <p>
+ * Public information for this service. A 3rd party registration could use this to register the service on a
+ * service registry.
+ * </p>
+ * <p>
+ * This info requires a name and a group of locatable services. A locatable service is
+ * a named group of {@link ServiceAcl}s.
+ * </p>
+ * <pre>
+ *     Map&lt;String, List&lt;ServiceAcl&gt;&gt; locatableServices = new HashMap&lt;&gt;();
+ *     List&lt;ServiceAcl&gt; helloAcls = Arrays.asList(
+ *        new ServiceAcl(Optional.of(Method.GET), Optional.of("?/hello/.*")),
+ *        new ServiceAcl(Optional.of(Method.POST), Optional.of("/login"))
+ *        );
+ *     List&lt;ServiceAcl&gt; goodbyeAcls = Arrays.asList(
+ *       new ServiceAcl(Optional.of(Method.POST), Optional.of("/logout/.*")));
+ *
+ *     locatableServices.put("hello-service", helloAcls);
+ *     locatableServices.put("goodbye-service", goodbyeAcls);
+ *
+ *     new ServiceInfo("GreetingService",locatableServices);
+ * </pre>
  */
 public final class ServiceInfo {
 
     private final String serviceName;
 
+    private final Map<String, List<ServiceAcl>> locatableServices;
+
+
+    /**
+     * @deprecated use {@link ServiceInfo#ServiceInfo(String, Map)} instead
+     */
     public ServiceInfo(String serviceName) {
```

Comment:
Nice, I didn't know the difference (JLS requiring warning when annotation is used). Always preferred `@deprecated` in the comments to add extra info. I wonder why Java's `@Deprecated` won't require info (like scala does).